source/slang/slang-ir-metal-legalize.cpp

```diff
@@ -1284,9 +1284,131 @@ namespace Slang
                     varLayoutBuilder.findOrAddResourceInfo(LayoutResourceKind::MetalPayload);
                     auto paramVarLayout = varLayoutBuilder.build();
                     builder.addLayoutDecoration(param, paramVarLayout);
+
+                    IRConstRefType* type = as<IRConstRefType>(param->getDataType());
+                    
+                    const auto annotatedPayloadType =
+                        builder.getPtrType(
+                            kIROp_ConstRefType,
+                            type->getValueType(),
+                            AddressSpace::MetalObjectData
+                        );
+
+                    param->setFullType(annotatedPayloadType);
                 }
             }
+            IROutputTopologyDecoration* outputDeco = entryPoint.entryPointFunc->findDecoration<IROutputTopologyDecoration>();
+            const auto topology = outputDeco->getTopology();
+            const auto topStr = topology->getStringSlice();
+            UInt topologyEnum = 0;
+            if(topStr.caseInsensitiveEquals(toSlice("point"))){
+                topologyEnum = 1;
+            } else if(topStr.caseInsensitiveEquals(toSlice("line"))) {
+                topologyEnum = 2;
+            } else if(topStr.caseInsensitiveEquals(toSlice("triangle"))) {
+                topologyEnum = 3;
+            } else {
+                // abort, idk how
```

Comment:
SLANG_UNEXPECTED("unknown topology");

Or you can just diagnose an error message with sink->diagnose(...)